core/textile/utils.go

```diff
@@ -23,3 +28,43 @@ func parseDbIDFromString(dbID string) (*thread.ID, error) {
 
 	return &id, nil
 }
+
+type deterministicThreadVariant string
+
+var (
+	metathreadThreadVariant deterministicThreadVariant = "metathread"
+)
+
+func newDeterministicThreadID(st *store.Store, threadVariant deterministicThreadVariant) (thread.ID, error) {
+	size := 32
+	variant := thread.Raw
+
+	kc := keychain.New(*st)
+
+	priv, _, err := kc.GetStoredKeyPairInLibP2PFormat()
+	if err != nil {
+		return thread.ID([]byte{}), err
+	}
+
+	privInBytes, err := priv.Raw()
+	if err != nil {
+		return thread.ID([]byte{}), err
+	}
+
+	num := pbkdf2.Key(privInBytes, []byte("threadID"+threadVariant), 256, size, sha512.New)
+	if err != nil {
+		return thread.ID([]byte{}), err
+	}
+
+	numlen := len(num)
+	// two 8 bytes (max) numbers plus num
+	buf := make([]byte, 2*binary.MaxVarintLen64+numlen)
+	n := binary.PutUvarint(buf, thread.V1)
+	n += binary.PutUvarint(buf[n:], uint64(variant))
+	cn := copy(buf[n:], num)
+	if cn != numlen {
+		panic("copy length is inconsistent")
+	}
+
+	return thread.ID(buf[:n+numlen]), nil
```

Comment:
Maybe we need more tests (or comments 😅 ) for this, just to be sure this is sufficient to generate a unique deterministic meta thread id per user.

It look to me like the `buf` buffer is always filled with the same set of value `thread.V1 + variant + numlen` and those values are always static and the same

Reply:
I think it's the `num` part that's unique across ID generations.  I think it's expected that the thread ID version and variant are the same while `num` which comes from the privkey is for the uniqueness.

Reply:
Oh, I just noticed it was `copy(..., num)` and not `copy(..., numlen)`. For some reason, I kept assuming numlen there when I was reviewing

Reply:
Added a test and comments in 4944baf :D